src/test/java/org/jabref/logic/cleanup/URLCleanupTest.java

```diff
@@ -0,0 +1,90 @@
+package org.jabref.logic.cleanup;
+
+import java.util.stream.Stream;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class URLCleanupTest {
+
+    @ParameterizedTest
+    @MethodSource("provideURL")
+    public void testChangeURL(BibEntry expected, BibEntry urlInputField) {
+        URLCleanup cleanUp = new URLCleanup();
+        cleanUp.cleanup(urlInputField);
+
+        assertEquals(expected, urlInputField);
+    }
+
+    private static Stream<Arguments> provideURL() {
+        /*
+         * Expected entries with various types of URLs(e.g, not secure protocol, password included for
+         * authentication, IP address, port etc.)
+         */
+        BibEntry[] expectedwithURL = {
+                new BibEntry().withField(StandardField.URL, "https://hdl.handle.net/10442/hedi/6089"),
+                new BibEntry().withField(StandardField.URL, "http://hdl.handle.net/10442/hedi/6089"),
+                new BibEntry().withField(StandardField.URL, "http://userid:password@example.com:8080"),
+                new BibEntry().withField(StandardField.URL, "http://142.42.1.1:8080"),
+                new BibEntry().withField(StandardField.URL, "http://☺.damowmow.com"),
+                new BibEntry().withField(StandardField.URL, "http://-.~_!$&'()*+,;=:%40:80%2f::::::@example.com"),
+                new BibEntry().withField(StandardField.URL, "https://www.example.com/foo/?bar=baz&inga=42&quux"),
+        };
+
+        // Expected entry in case note field holds more than two values
+        BibEntry expectedWithTwoNoteValues = new BibEntry()
+                .withField(StandardField.URL, "https://hdl.handle.net/10442/hedi/6089")
+                .withField(StandardField.NOTE, "this is a note");
+
+        // Expected entry in case note field holds more than one URLs
+        BibEntry expectedWithMultipleURLs = new BibEntry()
+                .withField(StandardField.URL, "https://hdl.handle.net/10442/hedi/6089")
+                .withField(StandardField.NOTE, "/url{http://142.42.1.1:8080}");
```

Comment:
Can you please test with `˜\url{}` ?  See also the link from koppor in the description. That would make more sense because it's a LaTeX command that is very liked to end up in the field.

Reply:
Done